src/Microsoft.PowerShell.Commands.Utility/commands/utility/select-object.cs

```diff
@@ -337,6 +337,11 @@ private void ProcessExpressionParameter()
             if (ExcludeProperty != null)
             {
                 _exclusionFilter = new MshExpressionFilter(ExcludeProperty);
+                if ((Property == null) || (Property.Length == 0))
+                {
```

Comment:
This will result a breaking change -- the scirpt below runs successfully before this change; but with this change a bunch errors will be thrown out.
```powershell
$s = Get-Process -Id $pid | Select-Object -ExcludeProperty ProcessName -ExpandProperty Modules
```
However, I don't feel this would break any existing scripts practically. It's documented that `-ExcludeProperty` has no effect without `-Property`, so I don't think any existing script would only specify `-ExcludeProperty`.

Maybe we can analyze the powershell scripts on GitHub to get some data. We can borrow @lzybkr's [AnalyzeCurlUsage.ps1](https://gist.github.com/lzybkr/3cd091334355f381d1d6ee7acfad5a48).